### DISCUSSION TITLE
Add setting to always automatically switch between the Script and 2D/3D editor instead of staying in Script Editor

### DISCUSSION THREAD

**Discussion by nubels** [godotengine/godot-proposals#7945] [Editor]
I think it would be a great UX improvement if there was an option to automatically switch between the Script and the 2D/3D editor instead of staying in the Script Editor when the .tscn file has a script attached to it. 

I find the current behaviour cumbersome in the following scenarious:

I am in the Script Editor and I want to modify another scene. I perform Quick Open Scene, select the scene I want to edit but it does not switch to the 2D/3D editor because the scene I selected has a script attached to it.

I am in the Script Editor and I want to open a scene via the FileSystem. I double click the .tscn file but I stay in the Script Editor because the scene I selected has a script attached to it.

**Comment by mieldepoche** [comment]
wasn't this the behavior in old versions? I think this was changed because precisely people were complaining that it was intrusive.
Btw you can setup shortcuts to change the displayed main view.

**Reply by nubels** [reply]
Thanks for the suggestions. I have shortcuts for that purpose. But this does not make the current behaviour it less cumbersome or unintuitive for me. When I select a .gd file, I want it to switch to the Script editor and when I select a .tscn file, I want it to switch to the 2D/3D editor without having to change it manually.

If there was a setting to change this behaviour which is turned off by default then everyone can change it to their liking.

I guess this shouldn't be too hard to implement if this was the behaviour in older versions...
